lib/optimizely/optimizely_factory.rb

```diff
@@ -19,37 +19,82 @@
-
-    def self.create_default_instance_with_config_manager(config_manager)
+    # Returns a new optimizely instance.
+    #
+    # @param config_manager - Required ConfigManagerInterface Responds to get_config.
+    def self.default_instance_with_manager(config_manager)
       Optimizely::Project.new(nil, nil, nil, nil, nil, nil, nil, config_manager)
     end
 
-    def self.create_default_instance(
-      datafile = nil,
+    # Returns a new optimizely instance.
+    #
+    # @params sdk_key - Required String uniquely identifying the datafile corresponding to project.
+    # @param fallback - Optional JSON string datafile.
+    # @param event_dispatcher - Optional EventDispatcherInterface Provides a dispatch_event method which if given a URL and params sends a request to it.
+    # @param logger - Optional LoggerInterface Provides a log method to log messages. By default nothing would be logged.
+    # @param error_handler - Optional ErrorHandlerInterface which provides a handle_error method to handle exceptions.
+    #                 By default all exceptions will be suppressed.
+    # @param skip_json_validation - Optional Boolean param to skip JSON schema validation of the provided datafile.
+    # @param user_profile_service - Optional UserProfileServiceInterface Provides methods to store and retreive user profiles.
+    # @param notification_center - Optional Instance of NotificationCenter.
+    def self.custom_instance(
+      sdk_key,
+      fallback = nil,
       event_dispatcher = nil,
       logger = nil,
       error_handler = nil,
       skip_json_validation = false,
       user_profile_service = nil,
-      sdk_key = nil,
-      config_manager = nil,
       notification_center = nil
     )
       Optimizely::Project.new(
-        datafile,
+        fallback,
         event_dispatcher,
         logger,
         error_handler,
         skip_json_validation,
         user_profile_service,
         sdk_key,
+        nil,
+        notification_center
+      )
+    end
+
+    # Returns a new optimizely instance.
+    #
+    # @param config_manager - Required ConfigManagerInterface Responds to get_config.
+    # @param event_dispatcher - Optional EventDispatcherInterface Provides a dispatch_event method which if given a URL and params sends a request to it.
+    # @param logger - Optional LoggerInterface which provides a log method to log messages. By default nothing would be logged.
+    # @param error_handler - Optional ErrorHandlerInterface which provides a handle_error method to handle exceptions.
+    #                 By default all exceptions will be suppressed.
+    # @param user_profile_service - Optional UserProfileServiceInterface which provides methods to store and retreive user profiles.
+    # @param skip_json_validation - Optional Boolean param to skip JSON schema validation of the provided datafile.
+    # @param notification_center - Optional Instance of NotificationCenter.
+    def self.custom_instance_with_manager(
```

Comment:
Maybe this is not needed. Let's keep it out until a customer asks for it. I don't see too much value in supporting this.